### PR TITLE
Issue-490: Fixes Featured content missing on Arabic.

### LINKED ISSUE #490: Featured content missing on Arabic [3h]
It seems to be related to the Slick carousel, as I can see the news for a sec, when moving to the next card

image

### PR DESCRIPTION
**Pull request by sonvir249**
### Things covered
1. Add `direction` variable `server_theme_carousel` theme hook
3. Added `direction` attribute to the slick slider parent.

### English Page:
image

### Arabic Page:
image

### Spanish Page:
image

### CODE REVIEW

File: web/modules/custom/server_general/src/ElementTrait/CarouselTrait.php [hunk 1]
```diff
@@ -41,14 +41,20 @@ protected function buildElementCarousel(string $title, array $body, array $items
...
     $elements[] = [
       '#theme' => 'server_theme_carousel',
       '#items' => $items,
       '#is_featured' => $is_featured,
       '#is_infinite' => $is_infinite,
     ];
 
+    $current_lang_code = \Drupal::languageManager()->getCurrentLanguage()->getId();
+    // RTL language codes.
+    $rtl_languages = ['ar', 'he', 'fa', 'ur'];
```
Comment (amitaibu) [inline web/modules/custom/server_general/src/ElementTrait/CarouselTrait.php:53]: You can get the lang object.

```php
current_lang_code = \Drupal::languageManager()->getCurrentLanguage();
```

Then you can keep

```php
$elements[] = [
      '#theme' => 'server_theme_carousel',
      '#items' => $items,
      '#is_featured' => $is_featured,
      '#is_infinite' => $is_infinite,
      '#dir' => $current-lang->... // Change here
    ];
```

File: web/modules/custom/server_general/src/ElementTrait/CarouselTrait.php [hunk 2]
```diff
@@ -41,14 +41,20 @@ protected function buildElementCarousel(string $title, array $body, array $items
       return [];
     }
 
-    $elements = [];
```
Comment (amitaibu) [inline web/modules/custom/server_general/src/ElementTrait/CarouselTrait.php:44]: Let's keep this.

**Review by amitaibu — COMMENTED** [review]
Thanks

File: web/modules/custom/server_general/src/ElementTrait/CarouselTrait.php (hunk 1 above)
Comment (sonvir249) [reply web/modules/custom/server_general/src/ElementTrait/CarouselTrait.php]: @amitaibu incorporate these changes.

File: web/modules/custom/server_general/src/ElementTrait/CarouselTrait.php (hunk 2 above)
Comment (sonvir249) [reply web/modules/custom/server_general/src/ElementTrait/CarouselTrait.php:44]: @amitaibu Sure.

File: web/modules/custom/server_general/src/ElementTrait/CarouselTrait.php [hunk 3]
```diff
@@ -41,12 +41,17 @@ protected function buildElementCarousel(string $title, array $body, array $items
...
+    // RTL language codes.
+    $rtl_languages = ['ar', 'he', 'fa', 'ur'];
+
     $elements = [];
     $elements[] = [
       '#theme' => 'server_theme_carousel',
       '#items' => $items,
       '#is_featured' => $is_featured,
       '#is_infinite' => $is_infinite,
+      '#dir' => in_array($current_lang_code->getId(), $rtl_languages) ? 'rtl' : 'ltr',
```
Comment (amitaibu) [inline web/modules/custom/server_general/src/ElementTrait/CarouselTrait.php:54]:
```suggestion
      '#dir' => $current_lang_code-> getDirection(),
```

File: web/modules/custom/server_general/src/ElementTrait/CarouselTrait.php [hunk 4]
```diff
@@ -41,12 +41,17 @@ protected function buildElementCarousel(string $title, array $body, array $items
       return [];
     }
 
+    $current_lang_code = \Drupal::languageManager()->getCurrentLanguage();
+    // RTL language codes.
+    $rtl_languages = ['ar', 'he', 'fa', 'ur'];
```
Comment (amitaibu) [inline web/modules/custom/server_general/src/ElementTrait/CarouselTrait.php:46]:
```suggestion
```

File: web/themes/custom/server_theme/server_theme.theme [hunk 5]
```diff
@@ -304,6 +304,8 @@ function server_theme_theme() {
       'is_infinite' => FALSE,
       // The render array of `server_theme_button`.
       'button' => NULL,
+      // Indicates the carousel direction.
+      'dir' => NULL,
```
Comment (amitaibu) [inline web/themes/custom/server_theme/server_theme.theme:308]:
```suggestion
      'direction' => NULL,
```

File: web/modules/custom/server_general/src/ElementTrait/CarouselTrait.php (hunk 3 above)
Comment (sonvir249) [reply web/modules/custom/server_general/src/ElementTrait/CarouselTrait.php]: @amitaibu this is done.

File: web/modules/custom/server_general/src/ElementTrait/CarouselTrait.php (hunk 4 above)
Comment (sonvir249) [reply web/modules/custom/server_general/src/ElementTrait/CarouselTrait.php]: @amitaibu this is done.

File: web/themes/custom/server_theme/server_theme.theme (hunk 5 above)
Comment (sonvir249) [reply web/themes/custom/server_theme/server_theme.theme]: @amitaibu this is done.

**Review by amitaibu — APPROVED** [review]
Thanks

**Pull request merged**